components/terraform/vault-deployment/values/vault-values.yaml

```diff
@@ -7,15 +7,18 @@ injector:
 server:
   image:
     repository: hashicorp/vault-enterprise
-    tag: 1.12.2-ent
+    tag: 1.13.1-ent
   dataStorage:
     enabled: true
     size: 1Gi
   auditStorage:
     enabled: true
     size: 1Gi
+  # authDelegator enables a cluster role binding to be attached to the service
+  # account.  This cluster role binding can be used to setup Kubernetes auth
+  # method.  https://www.vaultproject.io/docs/auth/kubernetes.html
   authDelegator:
-    enabled: false
+    enabled: true
```

Comment:
**Important:** authDelegator needs to be enabled, otherwise the vault-secrets-operator will just throw random permission denied on /login path of the Kubernetes auth method. Took me a few hours to notice this ;-)

See also https://developer.hashicorp.com/vault/docs/auth/kubernetes#how-to-work-with-short-lived-kubernetes-tokens for more information on the different options